x-pack/plugins/security_solution/server/endpoint/services/actions/action_list.ts

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger } from '@kbn/core/server';
+import type { ActionDetails, ActionListApiResponse } from '../../../../common/endpoint/types';
+import type { SecuritySolutionRequestHandlerContext } from '../../../types';
+
+import {
+  getActionsForAgents,
+  getActionResponses,
+  getTimeSortedActionDetails,
+} from '../../utils/action_list_helpers';
+
+import {
+  categorizeActionResults,
+  categorizeResponseResults,
+  getActionCompletionInfo,
+  mapToNormalizedActionRequest,
+} from './utils';
+
+interface GetActionListParam {
+  actionTypes?: string[];
+  context: SecuritySolutionRequestHandlerContext;
```

Comment:
Please remove this argument from the service.
A service should not be tied to an HTTP request context. I think the only thing you need from here is the Elastic serch client, so just have that as the required argument here (`esClient: ElasticsearchClient` )